components/services/bat_ledger/public/interfaces/bat_ledger.mojom

```diff
@@ -103,6 +103,7 @@ interface BatLedger {
   GetConfirmationsHistory(uint64 from_timestamp_seconds,
       uint64 to_timestamp_seconds) => (string transactions);
   GetRewardsInternalsInfo() => (string info);
+  DeleteStateFiles() => (bool success);
```

Comment:
These will not always be "Files". What we are asking Ledger to do is to reset the state so I think this method should be `ResetState()`. Also we shouldn't be using `bool` for success/failure, that's what ledger::Result is for

Reply:
I'm open to any naming convention suggestions as `ResetState()` seems to be used in quite a few places.

Also the end result determines whether or not to show the error so ultimately we would be return a bool `(result == LEDGER_OK)`

Reply:
the reason ledger::Result exists is to provide better information about failures. We should want consistency and that's why we use ledger::Result everywhere and shouldn't be using `bool`. if `result != LEDGER_OK` then how do you know what happened or how to attempt to resolve the problem if all you get is `false`?

Reply:
also `ResetState` is used in a few places and that's exactly what they are there for. We call `ResetState` on Ledger which in turn should be calling `ResetState` for everything it would normally call `LoadState`/'SaveState` on and then also calling db table delete methods

Reply:
`result != LEDGER_OK` is to report to the UI as a general failure to the user with "Try later" or "Try restarting and try again...". Sure, we can report back `ledger::DB_LOCKED` or something similar, but the resulting explanation to the user would be relatively the same.